services/agent/src/simcore_service_agent/_app.py

```diff
@@ -0,0 +1,102 @@
+import asyncio
+import logging
+import signal
+import traceback
+from collections import deque
+from contextlib import suppress
+from typing import Any, Awaitable, Callable, Coroutine, Final, Optional
+
+from pydantic import PositiveFloat, PositiveInt
+
+from ._meta import APP_FINISHED_BANNER_MSG, APP_STARTED_BANNER_MSG
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TASK_WAIT_ON_ERROR: Final[PositiveInt] = 10
+
+
+class Application:
```

Comment:
THOUGHT: I wonder if it would not have been better to start from the beginning with a ``FastAPI`` application even thought we do not need routes right away (probably we will soon!). We have a good understanding of all its parts and already provides a nice tools like events (``onstartup``, ``oncleanup``...), ``BackgroundTask``  and is launched by a server supervisor ``uvicorn``.

Check this out https://itnext.io/scheduler-with-an-api-rocketry-fastapi-a0f742278d5b

Reply:
moved to fastapi